lib/Dialect/TritonGPU/Transforms/Pipeliner/TC05MMAPipeline.cpp

```diff
@@ -570,20 +570,15 @@ void createBarrierAndWaitOps(IRRewriter &builder, scf::ForOp forOp,
   }
 }
 
-bool isSafeToPipeline(ttng::TCGen5MMAScaledOp scaledDot) {
-  auto isCopiedByTMEMCopy = [=](Value scale) {
-    if (!scale.hasOneUse()) {
-      // Should be used only by the scaled dot op
+bool isSafeToPipeline(ttng::TCGen5MMAScaledOp scaledDot, scf::ForOp forOp) {
+  // MMAv5 scaled dot (tcgen05.mma mxf8f6f4) is safe to be pipelined only
+  // when its scales in TMEM are stored by the TMEMCopy op (tcgen05.cp).
+  // That condition is equivalent to scale arguments of
+  // ttng::TCGen5MMAScaledOp being in SMEM during SWP in our convention.
+  auto isCopiedByTMEMCopy = [&](Value scale) {
+    auto scaleAlloc = findShmemAlloc(scale);
```

Comment:
I was missing the multi-buffering condition when I wrote that comment, a good point